### PR TITLE
Add systemd unit types.

### PR DESCRIPTION
**Pull request by pebenito**
Primarily contributed by Mike Palmiotto of the Tresys CLIP team.

### CODE REVIEW

File: shutdown.if [hunk 1]
```diff
@@ -91,6 +91,24 @@ interface(`shutdown_signal',`
...
 ## <summary>
+##     Send SIGCHLD signals to shutdown.
+## </summary>
+## <param name="domain">
+##     <summary>
+##     Domain allowed access.
+##     </summary>
+## </param>
+#
+interface(`shutdown_sigchld',`
```
Comment (perfinion) [inline shutdown.if]: why is this here? it seems unrelated to all the other changes

**Comment by perfinion** [comment]
There appear to be many places where there is a foo_initrc_exec_t but no associated foo_unit_t, will they be added later?

File: shutdown.if (hunk 1 above)
Comment (doverride) [inline shutdown.if]: -----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

On Mon, Oct 19, 2015 at 01:08:10PM -0700, Jason Zaman wrote:
> > @@ -91,6 +91,24 @@ interface(`shutdown_signal',`
> >  
> >  ########################################
> >  ## <summary>
> > +##     Send SIGCHLD signals to shutdown.
> > +## </summary>
> > +## <param name="domain">
> > +##     <summary>
> > +##     Domain allowed access.
> > +##     </summary>
> > +## </param>
> > +#
> > +interface(`shutdown_sigchld',`
> 
> why is this here? it seems unrelated to all the other changes

This is important. basically when you shutdown then shutdown_t becomes
pid 1

and since everyone needs to be able to send sighld to pid 1 they need to
also be able to send sigchld to shutdown_t.

thus this should be called in the domain.te file as:

tunable_systemd
    shutdown_sigchld(domain_t)
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/TresysTechnology/refpolicy-contrib/pull/4/files#r42419143

- -- 
02DFF788
4D30 903A 1CF3 B756 FB48  1514 3148 83A2 02DF F788
https://sks-keyservers.net/pks/lookup?op=get&search=0x314883A202DFF788
Dominick Grift
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQGcBAEBCgAGBQJWJVBNAAoJENAR6kfG5xmcRv0L/RYjCzOCV2glYm66hDC3940D
f659R1+Wi2tIAdmtkFXvFvHxj3w7eLqCoC0GoWJ7VjzSN1ZV5zxAT4rnWoiYx5NW
jcirX8sJz39XUW4LSgEo5tREcfgG4Lhfl7R0aWqDJzYZiUuVgFNbXm/1bsizcqlE
S228W1q0qkKubxNmPlKdI5YB02s8+2wftpAAJ0+CXWSrtOoMlrvn/yexMf+LA2Z2
v6k1xGF3Db3uHmIHAqo1NHDggAtgaQbB0Lax8PocR7Shuc/CkbA9EeGKq6uwf3Fz
W2BAb19rdYHqjgWjaxGk/EyoVxC88s6c6gSWyqUwZIVnVEQxdTpgFnq3Zls8gwSV
yT8gGKmNXTmJOY2+iZRueXjwQcGVDohUa3PxuSliaB5ryqr21N50qPf6d26MqI9q
X23Y3zKvzV6/k9dPnSMMpClhHcJ4SEozdujByvXQXehtUNUbXAZ904YLEpvb0gj/
YLfSIAmq10f+ce2/g9J7RZuGC2hEGHRkJWJQBexO9A==
=m77f
-----END PGP SIGNATURE-----

**Comment by doverride** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

On Mon, Oct 19, 2015 at 01:08:59PM -0700, Jason Zaman wrote:
> There appear to be many places where there is a foo_initrc_exec_t but no associated foo_unit_t, will they be added later?

Yes, probably go as the need arises. At this point low priority i
suppose.

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/TresysTechnology/refpolicy-contrib/pull/4#issuecomment-149331876

- -- 
02DFF788
4D30 903A 1CF3 B756 FB48  1514 3148 83A2 02DF F788
https://sks-keyservers.net/pks/lookup?op=get&search=0x314883A202DFF788
Dominick Grift
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQGcBAEBCgAGBQJWJVDsAAoJENAR6kfG5xmclXAMAJmgDnyioEc2Dp458UD1BqfR
H1vPrdIvR9wkfQQTV3fRwob1hPPr3Tv4HU3YpeKzRiREs7MohAQXvEdXJQVGyAUY
qXvK5d7J+y5bZGoa/8CbHt79Jd/WsGzJXYHflJ88pR2+aXt15ViGk3bYf4XVwloh
SGEa/OYR/HzfLfgVeqDUe8FNB3CDiDp65l8m+R6KmAERH/gRGsZeIMmStn9fp5CX
dNd1t7eEv6Za/qBb08uoZij0Eo1xMb0ZG1JSZrJyb9nTZoWAIxMOGc5qyNTBZq6N
PqBAPHXokVJJN7fFSnSjzIq83zZ13BK1EF6gIrxFjJ5TsBLLQUkaZiaGWX79mUp+
wA7V8eWUbZmkES1JHV0qBVYV3mg59CieFdszqNUwyj8Gh4/AK2FxLux07hZ+e2cB
ESRBlfpzF4l6ajLEdyQid+PQYqih+ySxIyuRyvDw9I91/HiMXbNbAXZQZn9AXjjn
lXIQVgl1nW9yV343rC25M4ctz+Dc60xyIW6xpwCYLw==
=hVaK
-----END PGP SIGNATURE-----

File: shutdown.if (hunk 1 above)
Comment (doverride) [inline shutdown.if]: err shutdown_sigchld(domain) , you get the point (i hope)

Comment (pebenito) [inline shutdown.if]: Makes sense.

**Comment by pebenito** [comment]
Yes, they will be added later as the need arises. Otherwise we'd have to go and install everything... and presumably not everything even has a unit. I suppose we could add the type and add labeling later.
